src/App.tsx

```diff
@@ -2,108 +1,77 @@
+import React, { useEffect, useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
-
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+import { Pagination } from './components/Pagination';
 
 export const App: React.FC = () => {
+  const totalItems: number = 42;
+  const values = [3, 5, 10, 20];
+
+  const [numItems, setNumItems] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const [from, setFrom] = useState(1);
+  const [to, setTo] = useState(numItems);
+
+  useEffect(() => {
+    let newFrom = 1;
+    let newTo = numItems;
+
+    if (currentPage) {
+      newFrom = (currentPage - 1) * numItems + 1;
+      newTo = Math.min(currentPage * numItems, totalItems);
+    }
+
+    setFrom(newFrom);
+    setTo(newTo);
+  }, [currentPage, numItems, totalItems]);
+
+  const items = getNumbers(from, to).map(n => `Item ${n}`);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${from} - ${to} of ${totalItems})`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
-            <option value="3">3</option>
-            <option value="5">5</option>
-            <option value="10">10</option>
-            <option value="20">20</option>
+            className="form-control"
+            value={numItems}
+            onChange={event => {
+              setNumItems(Number(event.target.value));
```

Comment:
It's a good practice to handle the conversion to a number as close to the event source as possible. Consider creating a separate handler function for the `onChange` event that encapsulates the number conversion logic.